### ISSUE TITLE
allow turning off syntax error diagnostics

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#874]
Avoid duplicate errors if the user is also using tsserver.

Requested by twitch.tv/DefineEvil.

**Comment by strager** [comment]
I think we can do this by changing how we categorize diagnostics.

Current categorization:
* error
* warning

Proposed categorization:
* early syntax error (e.g. missing parentheses after `if`)
* semantic error (e.g. duplicate const-declared variable)
* variable warnings (e.g. use of undeclared variable)
* other warning (e.g. use of `=` instead of `==`)

We would let users use editor settings to disable these categories. Or perhaps just a single checkbox, "avoid reporting errors detected by TypeScript".

This categorization might help with test262 too.

**Comment by strager** [comment]
@vegerot Do you use quick-lint-js with tsserver? What are your thoughts?

**Comment by vegerot** [comment]
Definitely disagree.  For syntax errors, quick-lint's diagnostics are far more helpful (and humorous).  I do agree it can be confusing seeing two diagnostics for one line, and having to look close at it to realize it's actually one problem.

<img width="1564" alt="image" src="https://user-images.githubusercontent.com/5297725/229180292-902f4e3d-4abf-47b5-8908-6bd3142d8483.png">

I'm not sure how to remove the confusion in these cases.  Of course for this specific problem quick-lint's and tsc's diagnostics are equally good, but in most places where they both report diagnostics, quick-lint's is better.

Normally, the way I tell whether there's really two problem or one problem is by looking at the error code 

<img width="912" alt="image" src="https://user-images.githubusercontent.com/5297725/229181677-0634eac4-83a3-4a98-90b9-6e94398933b7.png">


If I see one diagnostic starting with `E` and one starting with a number, I can easily tell it's just one problem.  Maybe one way to help would be making quick-lint's and error codes more distinct from tsserver's 🤔.  Or, do the opposite of this issue and make the quick-lint plugin hack the tsserver plugin and disable its redundant diagnostics!! 😈 

tl;dr: agree it is a _little_ confusing often having two diagnostics for the same error, but I like quick-lint's errors better

**Comment by vegerot** [comment]
> Proposed categorization:
> ...
> 
> We would let users use editor settings to disable these categories. Or perhaps just a single checkbox, "avoid reporting errors detected by TypeScript".


You could make a mapping of TypeScript error codes to quick-lint error codes, then have this checkbox suppress those diagnostics.

(if that's the path you want to go down.  As I said in my previous message, I like quick-lint's diagnostics better so I wouldn't enabled this option)

**Comment by vegerot** [comment]
https://twitter.com/samathingamajig/status/1648096828930289664?s=20
